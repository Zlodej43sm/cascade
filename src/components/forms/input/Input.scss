@use '../../../styles/colors';
@use '../../../styles/typography';
@use '../../../styles/sizes';

.input {
  @include typography.mulish(14px, 18px);
  padding: sizes.$base;
  box-sizing: border-box;
  width: 100%;
  background-color: colors.$grey-10;
  border: none;
  border-radius: 5px;
  color: colors.$base-dark;
}
