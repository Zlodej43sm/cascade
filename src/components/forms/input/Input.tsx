import React from 'react';
import clsx from 'clsx';

import styles from './Input.scss';

export type InputProps = Pick<React.HTMLProps<HTMLInputElement>, 'id' | 'name' | 'value' | 'placeholder' | 'className' | 'onChange' | 'onBlur'>;

function Input({ className, ...rest }: InputProps) {
  return (
    <input className={clsx(styles.input, className)} {...rest} />
  );
}

export default Input;
