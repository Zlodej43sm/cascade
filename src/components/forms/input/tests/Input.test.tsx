import React from 'react';
import { render } from '@testing-library/react';

import Input from '../Input';

describe('Input', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
});
