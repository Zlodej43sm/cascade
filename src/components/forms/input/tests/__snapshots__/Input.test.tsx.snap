// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="input"
  />
</DocumentFragment>
`;
